src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+const elements = [...document.querySelectorAll('li')];
+
+function getEmployees(employees) {
+  return employees.map((employee) => ({
+    name: employee.textContent.trim(),
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: employee.dataset.age,
+    element: employee,
+  }));
+}
+
+function sortList(list) {
+  const sortedEmployees = [...list].sort((employee1, employee2) => {
+    const salary1 = parseInt(employee1.salary.replace(/[^0-9]/g, ''), 10);
+    const salary2 = parseInt(employee2.salary.replace(/[^0-9]/g, ''), 10);
+
+    return salary2 - salary1;
+  });
+
+  const listContainer = document.querySelector('ul');
```

Comment:
Make sure that there is only one 'ul' element in the document or that this selector targets the correct list container. If there are multiple 'ul' elements, consider using a more specific selector.